actions/smoke-test

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+set -ex
+
+if ! charms.reactive is_state 'spark.started'; then
+    action-fail 'Spark not yet ready'
+    exit
+fi
+
+NUM_PARTITIONS=10
+
+# create dir to store results
+RUN=`date +%s`
+RESULT_DIR=/opt/sparkpi-results
+RESULT_LOG=${RESULT_DIR}/${RUN}.log
+mkdir -p ${RESULT_DIR}
+chown -R ubuntu:ubuntu ${RESULT_DIR}
+
+START=`date +%s`
+# NB: Escaped vars in the block below (e.g., \${MASTER}) come from
+# /etc/environment while non-escaped vars (e.g., ${NUM_PARTITIONS}) are
+# parameterized from this outer scope
+su ubuntu << EOF
+. /etc/environment
+echo 'calculating pi'
+spark-submit --master \${MASTER} --class org.apache.spark.examples.SparkPi \
+  /usr/lib/spark/lib/spark-examples-*.jar ${NUM_PARTITIONS} &> ${RESULT_LOG}
+EOF
+STOP=`date +%s`
+
+DURATION=`expr $STOP - $START`
```

Comment:
do we need to action-set any output here?  like "pi is roughly 3.x" from the output of spark-submit, or is that somehow implicitly returned in the action fetch output?